chain-impl-mockchain/src/tokens.rs

```diff
@@ -0,0 +1,208 @@
+use chain_core::mempack::{ReadBuf, ReadError, Readable};
+use cryptoxide::{blake2b::Blake2b, digest::Digest};
+use thiserror::Error;
+use typed_bytes::ByteBuilder;
+
+use std::convert::TryInto;
+
+pub const POLICY_HASH_SIZE: usize = 28;
+pub const TOKEN_NAME_MAX_BYTES: usize = 32;
+
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+pub struct TokenIdentifier {
+    pub policy_hash: PolicyHash,
+    pub token_name: TokenName,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct PolicyHash([u8; POLICY_HASH_SIZE]);
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct TokenName(Vec<u8>);
+
+/// A minting policy consists of multiple entries defining different
+/// constraints on the minting process. An empty policy means that new tokens
+/// cannot be minted during the chain run.
```

Comment:
saying that a minting policy is a constraint for minting and that no minting policies means no minting is a bit of a contradiction, but as long as it's clearly specified I'm ok with that